src/WPConfigTransformer.php

```diff
@@ -226,8 +226,21 @@ public function remove( $type, $name ) {
 			return false;
 		}
 
-		$pattern  = sprintf( '/(?<=^|;|<\?php\s|<\?\s)%s\s*(\S|$)/m', preg_quote( $this->wp_configs[ $type ][ $name ]['src'], '/' ) );
-		$contents = preg_replace( $pattern, '$1', $this->wp_config_src );
+		if ( 'constant' === $type ) {
+			$pattern = sprintf(
+				"/\bdefine\s*\(\s*['\"]%s['\"]\s*,\s*(('[^']*'|\"[^\"]*\")|\s*(?:[\s\S]*?))\s*\)\s*;\s*/mi",
+				preg_quote( $name, '/' )
+			);
+		} elseif ( 'variable' === $type ) {
+			$pattern = sprintf(
+				'/^\s*\$%s\s*=\s*[\s\S]*?;\s*$/mi',
+				preg_quote( $name, '/' )
+			);
+		} else {
+			return false;
```

Comment:
This `return false` seems redundant to me. If not valid type then it will return false earlier in 226. Thoughts?

Reply:
Yes, you're right. If the type isn't 'constant' or 'variable', the function won't go past the initial check at line 226. Handling 'constant' in the `if` condition and 'variable' in the `else` condition should cover all our needs without extra checks.

Does this sound good to you?